src/bin/pg_autoctl/file_utils.h

```diff
@@ -24,6 +24,21 @@
 #endif
 
 
+/*
+ * In order to avoid dynamic memory allocations and tracking when searching the
+ * PATH environment, we pre-allocate 1024 paths entries. That should be way
+ * more than enough for all situations, and only costs 1024*1024 = 1MB of
+ * memory.
+ */
+typedef struct SearchPath
+{
+	int total;
+	int found;
+	char entries[1024][MAXPGPATH];
```

Comment:
It seems only `matches` is actually used outside of the `search_path` function. So I think this can be removed from the struct (as well as the `total`).

Reply:
Fair point. I wanted it to be more generic that our needs, which is seldom a good idea. Fixed now.